packages/site/src/lib/components/Agenda.svelte

```diff
@@ -11,6 +11,16 @@
 
   export let events: Event[] = [];
   export let title: string;
+  
+  const maxPerPage: number = 5;
+
+  let pageNumber: number = 1;
+  let filteredEvents: Event[] = events;
+  let numPages: number = Math.ceil(filteredEvents.length / maxPerPage);
+
+  let currentPageEvents: Event[];
+
+  let searchValue: string = '';
```

Comment:
```suggestion
  let pageNumber: number = 1;
  let searchValue: string = '';
  
  const eventMatches = (ev: Event, search: string): boolean => ev.name.toLowerCase().includes(search.toLowerCase())
  const filterEvents = (evs: Event[], search: string): Event[] => search.trim() === '' ? evs : evs.filter(ev => eventMatches(ev, search))
  
  $: filteredEvents: Event[] = filterEvents(events, searchString)
  
  const paginateEvents = (evs: Event[], pageNum: number, maxPerPage: number): Event[] => {
    const start = (pageNum - 1) * maxPerPage
    const end = start + maxPerPage
    return evs.slice(start, end)
  }
  
  $: currentPageEvents: Event[] = paginateEvents(filteredEvents, pageNumber, maxPerPage)
  
  $: numPages: number = Math.ceil(filteredEvents.length / maxPerPage);
```

This should use the reactive declaration syntax. Saves you a toooon of work. You have to explicitly reference the variables in the statement or block. So the reason why I pass all the variables as arguments to `paginateEvents` and `filterEvents` is to make sure each of the `$` blocks knows its dependencies.